### ISSUE TITLE
Compilation Issues if JSON Node Name has Space

### ISSUE THREAD

**Issue by sromanuk** [bluelinelabs/LoganSquare#76]
Hi! I have an issue with the library. 
Mine JSON has node like that:
```
"Body Part": [
    {
       ...
    }
]
```

I can't change it, so have to find a workaround to make it work.
During compilation I'm receiving this error message:
```
Class$$JsonObjectMapper.java:88: error: ';' expected
    final List<BodyPart> lslocalBody Part = object.getBodyParts();
                                    ^
Class$$JsonObjectMapper.java:89: error: ')' expected
    if (lslocalBody Part != null) {
                   ^
Class$$JsonObjectMapper.java:89: error: not a statement
    if (lslocalBody Part != null) {
                         ^
Class$$JsonObjectMapper.java:89: error: ';' expected
    if (lslocalBody Part != null) {
                                ^
Class$$JsonObjectMapper.java:92: error: ')' expected
      for (BodyPart element1 : lslocalBody Part) {
                                          ^
Class$$JsonObjectMapper.java:92: error: not a statement
      for (BodyPart element1 : lslocalBody Part) {
                                           ^
Class$$JsonObjectMapper.java:92: error: ';' expected
      for (BodyPart element1 : lslocalBody Part) {
                                               ^
7 errors
```

My code looks like that:
```
@JsonObject
public class Workout implements Parcelable {
    ...
    private static final String FIELD_BODY_PART = "Body Part";
    
    @JsonField(name = FIELD_BODY_PART)
    private List<BodyPart> mBodyParts;
    ...
}
```

I'm going to dive into your codebase on weekends but really hope you'll fix this issue faster that me :)
In any case, thanks for that library!

**Comment by aurae** [comment]
This is a fairly easy fix. We need to perform some cleanup on field names in the processor's Type implementations for Lists. BTW, the same bug exists for Map fields as well. I'm about to submit a PR that fixes this.

**Comment by sromanuk** [comment]
@aurae What if JSON will have two nodes: `two words` and `two  words` or `twowords`. I know, it's crazy, but if it happens the we'll have a problem. Maybe for the future release this changes might be adopted for that situation too.

**Comment by aurae** [comment]
The former case you described wouldn't be RFC7159-compliant, however both cases should still produce compileable code, even if they are quite unlikely.

I have updated my PR, so that the ObjectMapperInjector keeps track of which field names have already been processed. It passes this information down to the individual field holders, which use that info to produce unique variable names.

This model:
```java
@JsonField(name = "Address Lines")
public List<String> addressLines;

@JsonField(name = "AddressLines")
public List<String> addressLinesDuplicate;
```

now produces this output:
```java
final List<String> lslocalAddressLines = object.addressLines;
if (lslocalAddressLines != null) {
  jsonGenerator.writeFieldName("Address Lines");
  jsonGenerator.writeStartArray();
  for (String element1 : lslocalAddressLines) {
	if (element1 != null) {
	  jsonGenerator.writeString(element1);
	}
  }
  jsonGenerator.writeEndArray();
}
final List<String> lslocalAddressLines1 = object.addressLinesDuplicate;
if (lslocalAddressLines1 != null) {
  jsonGenerator.writeFieldName("AddressLines");
  jsonGenerator.writeStartArray();
  for (String element1 : lslocalAddressLines1) {
	if (element1 != null) {
	  jsonGenerator.writeString(element1);
	}
  }
  jsonGenerator.writeEndArray();
}
```

**Comment by EricKuck** [comment]
This is now merged into 1.2.5. Thanks @aurae!
